extensions/ql-vscode/src/remote-queries/remote-queries-manager.ts

```diff
@@ -61,6 +65,13 @@ export class RemoteQueriesManager {
     query: RemoteQuery,
     cancellationToken: CancellationToken
   ): Promise<void> {
+    const queryId = this.createQueryId(query.queryName);
```

Comment:
Yes...the next PR will be about refactoring how we do monitoring and storing the query. I'd like to centralize the object that (de-)serializes query data, so this can happen outside of the initial running of the query.